File: adcs-backend/app/schemas/translation.py
from pydantic import BaseModel, ConfigDict
from typing import List, Optional
from uuid import UUID
from datetime import datetime

class TranslationResponse(BaseModel):
    id: UUID
    filename: str
    file_type: str
    status: str
    comment: Optional[str] = None
    result_file_url: Optional[str] = None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)

class UploadTranslationResponse(BaseModel):
    message: str
    duplicate_count: int
    processed_files: List[TranslationResponse]

class CommentUpdateRequest(BaseModel):
    comment: str

File: adcs-frontend/src/api/translation.ts
import axiosClient from "./axiosClient";

export type FileStatus = 'pending' | 'translating' | 'success' | 'failed';

export interface TranslationFile {
  id: string;
  filename: string;
  file_type: string;
  status: FileStatus;
  comment: string;
  result_file_url?: string;
  created_at: string;
}

export interface UploadTranslationResponse {
  message: string;
  duplicate_count: number;
  processed_files: TranslationFile[];
}

// Gửi file lên backend (Sử dụng FormData)
export async function uploadTranslationFiles(files: File[]): Promise<UploadTranslationResponse> {
  const formData = new FormData();
  files.forEach(file => {
    formData.append("files", file); 
  });
  
  return axiosClient.post(`/translations/upload`, formData);
}

// Lấy danh sách lịch sử dịch
export async function getTranslations(): Promise<TranslationFile[]> {
  return axiosClient.get(`/translations/`);
}

// Cập nhật nhận xét
export async function updateTranslationComment(id: string, comment: string): Promise<{ message: string }> {
  return axiosClient.put(`/translations/${id}/comment`, { comment });
}